lib/OpenQA/WebAPI/Auth/OAuth2.pm

```diff
@@ -93,7 +93,8 @@ sub auth_login ($controller) {
     croak 'Config was not parsed' unless my $main_config = $controller->app->config->{oauth2};
     croak 'Setup was not called' unless my $provider_config = $main_config->{provider_config};
 
-    my $get_token_args = {redirect_uri => $controller->url_for('login')->userinfo(undef)->to_abs};
+    my $url = $controller->app->config->{global}->{base_url} || $controller->req->url->base;
+    my $get_token_args = {redirect_uri => "${url}/login"};
```

Comment:
Maybe the following works and would preserve the use of `url_for` to find the URL for that named route:
```suggestion
    my $base_url = $controller->app->config->{global}->{base_url};
    my $host = ($base_url ? Mojo::URL->new($base_url) : $controller->req->url)->host;
    my $get_token_args = {redirect_uri => "$controller->url_for('login')->host($host)->userinfo(undef)};
```

Reply:
that had a few errors in it, but with those fixed, yeah, seems to work.